# DP: Apply proposed patch for PR target/67280 (ARM).

gcc/ChangeLog:

<DATE>  Charles Baylis  <charles.baylis@linaro.org>

	* cgraphunit.c (cgraph_node::create_wrapper): Set can_throw_external
	in new callgraph edge.

Change-Id: I13423872e37f3ca8e0bd61e8eac4a06e8e107a2c
---
 gcc/cgraphunit.c | 1 +
 1 file changed, 1 insertion(+)

Index: b/src/gcc/cgraphunit.c
===================================================================
--- a/src/gcc/cgraphunit.c
+++ b/src/gcc/cgraphunit.c
@@ -2505,6 +2505,7 @@ cgraph_node::create_wrapper (cgraph_node
   memset (&thunk, 0, sizeof (cgraph_thunk_info));
   thunk.thunk_p = true;
   create_edge (target, NULL, count, CGRAPH_FREQ_BASE);
+  callees->can_throw_external = !TREE_NOTHROW (target->decl);
 
   tree arguments = DECL_ARGUMENTS (decl);
 
